qiskit_research/utils/gates.py

```diff
@@ -136,6 +136,44 @@ def __array__(self, dtype=None):
         return np.array([[0, -1j], [1j, 0]], dtype=dtype)
 
 
+class PiPhiGate(Gate):
+    r"""The single-qubit X gate (:math:`\sigma_x`), implemented
+    with an angle :math:`\\phi` relative to the X-axis.
```

Comment:
Calling it "the X gate" seems a bit misleading. How about something like "a 90-degree rotation about an axis offset by angle phi from the X axis in the X-Y plane"? Also, please follow the convention of having a one-line short docstring followed by a more detailed explanation if necessary.

Reply:
Done!